### DISCUSSION TITLE
ModuleNotFoundError: No module named 'airflow.providers.amazon.aws.operators.emr_add_steps'

### DISCUSSION THREAD

**Discussion by ThinSnorlax** [apache/airflow#32264] [Q&A] [answered]
Hi Team

I create a airflow on aws, and it's version is 2.5.1, and I create a dag but it can't import to airflow;
dags:
```
from datetime import datetime, timedelta
from airflow import DAG
from airflow.providers.amazon.aws.operators.emr_add_steps import EmrAddStepsOperator
from airflow.providers.amazon.aws.sensors.emr_step import EmrStepSensor
import common_utils

default_args = {
    'owner': 'airflow',
    'depends_on_past': False,
    'start_date': datetime(2023, 6, 20),
    'email': ['edward.hu@trendsi.com'],
    'retries': 1,
    'retry_delay': timedelta(minutes=5),
}

dag = DAG(
    'emr_hive_dag',
    default_args=default_args,
    description='Submit Hive job to EMR cluster',
    schedule_interval='0 0 * * *',
    catchup=False
)


hive_script = """ 
     select * from ods.test; """

step_adder = EmrAddStepsOperator(
    task_id='add_hive_step',
    job_flow_id='EMR cluster id ***',
    aws_conn_id='aws-us-east-2',
    steps=[
        {
            'Name': 'Hive Job',
            'ActionOnFailure': 'CONTINUE',
            'HadoopJarStep': {
                'Jar': 'command-runner.jar',
                'Args': ['hive',  '-f', 'Hive script path']
            },
        },
    ],
    dag=dag
)


step_sensor = EmrStepSensor(
    task_id='hive_step_sensor',
    job_flow_id='EMR cluster id ***',
    step_id="{{ task_instance.xcom_pull('add_hive_step', key='return_value')[0]}}",
    aws_conn_id='aws-us-east-2',
    dag=dag
)
```

I have install the [apache-airflow-providers-amazon], like the below of:

Package Name | Version | Description
-- | -- | --
apache-airflow-providers-amazon | 8.2.0 | Amazon integration (including Amazon Web Services (AWS)).

Package Name	Version	Description
[apache-airflow-providers-amazon](https://airflow.apache.org/docs/apache-airflow-providers-amazon/8.2.0/)	8.2.0	Amazon integration (including [Amazon Web Services (AWS)](https://aws.amazon.com/)).

My airflow is Amazon MWAA

**Comment by potiuk** [comment, marked as answer]
Where did you get the import from? It seems to not be there, so I am not sure why you expect it to be there https://airflow.apache.org/docs/apache-airflow-providers-amazon/8.2.0/_api/airflow/providers/amazon/index.html ?

Did you check the python API for the operator?

**Reply by ThinSnorlax** [reply]
Thank you very much, I find the legacy version doc: https://airflow.apache.org/docs/apache-airflow-providers-amazon/1.0.0/_api/airflow/providers/amazon/aws/operators/emr_add_steps/index.html (version 1.0.0)
and the function had been remove in 7.1.0
